articles/quickstart/webapp/nodejs/01-login.md

```diff
@@ -14,4 +16,247 @@ github:
+
+* [passport](http://www.passportjs.org/) - an authentication middleware for Node.js
+* [passport-auth0](https://github.com/auth0/passport-auth0) - an Auth0 authentication strategy for Passport
+* [express-session](https://www.npmjs.com/package/express-session) - a middleware to manage sessions
+* [connect-ensure-login](https://github.com/jaredhanson/connect-ensure-login) - a middleware to ensure a user must be logged in, in order to access certain routes
+
+```bash
+# installation with npm
+npm install passport passport-auth0 express-session connect-ensure-login --save
+```
+
+or
+
+```bash
+# installation with yarn
+yarn add passport passport-auth0 express-session connect-ensure-login
+```
+
+### Configure express-session
+
+In `app.js`, include the `express-session` module and configure it. The `secret` parameter is a secret string that is used to sign the session ID cookie. Please use a custom value.
+
+```js
+// app.js
+
+var session = require('express-session');
+
+//session-related stuff
+var sess = {
+ secret: 'CHANGE THIS SECRET',
+ cookie: {},
+ resave: false,
+ saveUninitialized: true
+};
+
+if (app.get('env') === 'production') {
+ sess.cookie.secure = true; // serve secure cookies, requires https
+}
+
+app.use(session(sess));
+```
+
+### Configure Passport with the application settings
+
+In `app.js`, include the `passport` and `passport0-auth0` modules, and configure Passport to use a new instance of `Auth0Strategy` with your Auth0 application settings. Use `passport.initialize()` and `passport.session()` to initialize Passport with persistent login sessions.
+
+```js
+// app.js
+var Auth0Strategy = require('passport-auth0'),
+    passport = require('passport');
+
+//passport-auth0
+var strategy = new Auth0Strategy({
+  domain: '${account.namespace}',
+  clientID: '${account.clientId}',
+  clientSecret: 'YOUR_CLIENT_SECRET', // Replace this with the client secret for your app
+  callbackURL: 'http://localhost:3000/callback'
+ },
+ function(accessToken, refreshToken, extraParams, profile, done) {
+   // accessToken is the token to call Auth0 API (not needed in the most cases)
+   // extraParams.id_token has the JSON Web Token
+   // profile has all the information from the user
+   return done(null, profile);
+ }
+);
+
+passport.use(strategy);
+
+app.use(passport.initialize());
+app.use(passport.session());
+```
+
+### Storing and retrieving user data from the session
+
+In a typical web application, the credentials used to authenticate a user are only transmitted during the login request. If authentication succeeds, a session is established and maintained via a cookie set in the user's browser. Each subsequent request does not contain credentials, but rather the unique cookie that identifies the session.
+
+To support login sessions, Passport serializes and deserializes user instances to and from the session. Optionally, you may want to serialize only a subset to reduce the footprint, i.e., `user.id`.
```

Comment:
I do not know Passport, but this section is still unclear to me - what is it doing by default, and why would I want to change it, and is the below an example of me changing it? It looks like the example is just serializing `user`  - is that already happening, or do I need to add the below snippet to my `app.js`?